lib/daru/io/importers/redis.rb

```diff
@@ -0,0 +1,176 @@
+        #   # Key "age"    => [32, 37, 37, 19]
+        #   # Key "living" => [true, true, true, false]
+        #
+        #   # Say `connection` is a Redis instance
+        #   df         = Daru::DataFrame.from_redis(connection, "name", "age")
+        #
+        #   df
+        #
+        #   #=> <Daru::DataFrame(4x2)>
+        #   #=>           name     age
+        #   #=>      0  Tyrion      32
+        #   #=>      1   Jamie      37
+        #   #=>      2  Cersei      37
+        #   #=>      3 Joffrey      19
+        #
+        # @example Querying for matching keys with count
+        #   # Say, the Redis connection has this setup
+        #   # Key "timestamp:100620171225" => { "name" => "Joffrey", "age" => 19 }.to_json
+        #   # Key "timestamp:090620171222" => { "name" => "Cersei",  "age" => 37 }.to_json
+        #   # Key "timestamp:090620171218" => { "name" => "Jamie",   "age" => 37 }.to_json
+        #   # Key "timestamp:090620171216" => { "name" => "Tyrion",  "age" => 32 }.to_json
+        #
+        #   # Say `connection` is a Redis instance
+        #   df = Daru::DataFrame.from_redis(connection, match: "timestamp:09*", count: 3)
+        #
+        #   df
+        #
+        #   #=> <Daru::DataFrame(3x2)>
+        #   #=>                               name     age
+        #   #=>      timestamp:090620171222  Cersei      37
+        #   #=>      timestamp:090620171218   Jamie      37
+        #   #=>      timestamp:090620171216  Tyrion      32
+        def initialize(connection={}, *keys, match: nil, count: nil,
+          page: 0)
+          @match  = match
+          @count  = count
+          @page   = page
+          @client = get_client(connection)
+          @keys   = choose_keys(*keys).map(&:to_sym)
+        end
+
+        def call
+          vals = @keys.map { |key| ::JSON.parse(@client.get(key), symbolize_names: true) }
+          Util.guess_parse @keys, vals
+        end
+
+        private
+
+        def choose_keys(*keys)
+          if keys.count.zero?
+            cursor = 0
+            page   = 0
+            cursor, keys = @client.scan(cursor, match: @match, count: @count)
+            return keys.uniq if @page.zero?
+
+            # Loop to iterate through paginated results of Redis#scan
+            # If @pages < 0 or @pages > total_pages, fetch ALL keys.
+            # Else, fetch the particular page of keys.
+            until cursor.to_i.zero?
+              response = @client.scan(cursor, match: @match, count: @count)
+              return response.last.uniq if @page == (page+=1)
+              cursor = response.first.to_i
+              keys  += response.last
+            end
+            keys.uniq
+          else
+            keys.to_a
+          end
+        end
+
+        def get_client(connection)
+          if connection.is_a? ::Redis
+            connection
+          elsif connection.is_a? Hash
+            ::Redis.new connection
+          end
```

Comment:
1. `case` looks more appropriate here
2. ...and if it is neither `Redis` nor `Hash`?..